### DISCUSSION TITLE
Self updating question

### DISCUSSION THREAD

**Discussion by TheCableGuy99** [007revad/Synology_HDD_db#316] [General]
I have a task setup for on boot to update this script with the command:
/volume1/syno_hdd_db.sh

Now as there was a recent update I just thought I'd run it manually to udpate. The issue is I was prompted for an action to update it:

```
root@SHS1:~# /volume1/syno_hdd_db.sh
Synology_HDD_db v3.5.91
DS1821+ DSM 7.2.1-69057-5
StorageManager 1.0.0-0017

ds1821+_host_v7 version 8054

Running from: /volume1/syno_hdd_db.sh

There is a newer version of this script available.
Current version: v3.5.91
Latest version:  v3.5.93
Do you want to download v3.5.93 now? [y/n]
```

So if it requires manual intervention how do I automate the update? Can I add "--yes" or something similar? 

Thanks.

**Comment by 007revad** [comment]
You want to use the --autoupdate= option

When scheduled I would use `syno_hdd_db.sh -ne --autoupdate=3`

```
  -n, --noupdate        Prevent DSM updating the compatible drive databases
  -e, --email           Disable colored text in output scheduler emails
      --autoupdate=AGE  Auto update script (useful when script is scheduled)
                          AGE is how many days old a release must be before
                          auto-updating. AGE must be a number: 0 or greater
```

**Comment by TheCableGuy99** [comment]
Ahhh thank you Dave.

Just to confirm, you are not saying use -ne as well? You included that just for reference, right?

So right now i have:
/volume1/syno_hdd_db.sh --autoupdate=2

Thats okay right?

Thanks again!

**Reply by 007revad** [reply]
You should always use -n to prevent DSM updating the drive database and undoing the changes the script made.

When the script is scheduled you should use -e to prevent the task scheduler emails and task scheduler's "Action > View Result" from containing gibberish like `\e[0;36m` where in SSH you'd see colored text.

**Comment by TheCableGuy99** [comment]
ahh okay, thank you for explanation. I've updated as per your example.

Many thanks!
